src/pages/journal/components/journal-navigation/journal-navigation.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { StoreModel } from '../../../../common/store.model';
+import { getJournalState } from '../../journal.reducer';
+import { map } from 'rxjs/operators';
+import { getSelectedDate, canNavigateToPreviousDay, canNavigateToNextDay, isSelectedDateToday } from '../../journal.selector';
+import { Observable } from 'rxjs/Observable';
+
+import { SelectPreviousDay, SelectNextDay } from '../../journal.actions';
+
+interface JournalNavigationPageState {
+  selectedDate$: Observable<string>,
+  canNavigateToPreviousDay$: Observable<boolean>,
+  canNavigateToNextDay$: Observable<boolean>,
+  isSelectedDateToday$: Observable<boolean>,
+}
+
+@Component({
+  selector: 'journal-navigation',
+  templateUrl: 'journal-navigation.html'
+})
+export class JournalNavigationComponent implements OnInit {
+
+  pageState: JournalNavigationPageState;
+
+  constructor(private store$: Store<StoreModel>) {}
+
+  ngOnInit(): void {
+    this.pageState = {
+      selectedDate$: this.store$.pipe(
+        select(getJournalState),
+        map(getSelectedDate)
+      ),
+      canNavigateToPreviousDay$: this.store$.pipe(
+        select(getJournalState),
+        map(canNavigateToPreviousDay)
+      ),
+      canNavigateToNextDay$: this.store$.pipe(
+        select(getJournalState),
+        map(canNavigateToNextDay)
+      ),
+      isSelectedDateToday$: this.store$.pipe(
+        select(getJournalState),
+        map(getSelectedDate),
+        map(isSelectedDateToday)
+      )
+    };
+  }
+
+  onPreviousDayClick() {
+    this.store$.dispatch(new SelectPreviousDay());
+  }
+
+  onNextDayClick() {
```

Comment:
can you add :void for the return type for this and the previous method. To keep it consistent with ngOnInit Method

Reply:
Added